src/templates/home-page.js

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from "react";
+import { render } from "react-dom";
+
+const HomeTile = ({ title, body, iconName }) => <div className="home-tile">
+  <i className={`icon fa-light fa-${iconName}`}></i>
+  <h4 className="title">{title}</h4>
+  <div className="body" dangerouslySetInnerHTML={{ __html: body }} />
+</div>
+
+const StartTour = (text) => <a
+  href="#"
+  className="btn btn-default"
+>
+  {text || "Start Tour"}
+</a>
+
+const HomePage = ({tiles}) => {
+  useEffect(() => {
+    const previewPane = document.getElementById("preview-pane").contentDocument
+    previewPane.querySelectorAll("start-tour").forEach((el) => {
+      render(StartTour(el.textContent), el)
+    });
+  });
```

Comment:
```suggestion
  }, []);
```
This useEffect is missing a dependency array, and I think you're better off using createPortal() instead of render().

Two reasons: a) `render(<Foo />)` is being deprecated in favor of `createRoot(el).render(<Foo />)` and b) nesting a React app inside another React app is icky. createPortal() will give you the same placement, but features like React Context will work across the portal boundary.

Also your StartTour component doesn't look like it actually triggers the tour.

```ts
import useBackboneContext from "host/useBackboneContext"

function StartTourButton({ children="Start Tour" }) {
  const { dispatcher } = useBackboneContext()
  const handler = () => dispatcher.trigger("show:tour")

  return (
    <button className="btn btn-default" onClick={handler}>{children}</button>
  )
}
```

Reply:
(I'll check out createPortal; I agree nested rendering is icky, but I wasn't aware there was an alternative.)